packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -218,8 +229,12 @@ export class Metric implements IMetric {
     }
     if (props.dimensions) {
       this.validateDimensions(props.dimensions);
+      this.dimensions = props.dimensions;
+    }
+    if (props.dimensionsMap) {
+      this.validateDimensions(props.dimensionsMap);
+      this.dimensions = props.dimensionsMap;
     }
-    this.dimensions = props.dimensions;
```

Comment:
Small suggestion: if you change `validateDimensions()` to return its argument after a success, and handle the `undefined` case, this entire blocks simplifies to:

```suggestion
    this.dimensions = this.validateDimensions(props.dimensionsMap ?? props.dimensions),
```